src/gyroaverage/gyroaverage_operator.hpp

```diff
@@ -0,0 +1,210 @@
+ * @tparam BSplinesTheta       The B-spline basis for the theta direction.
+ */
+template <
+        class ExecutionSpace,
+        class GridR,
+        class GridTheta,
+        class GridBatch,
+        class BSplinesR,
+        class BSplinesTheta>
+class GyroAverageOperator
+{
+    /**
+     * @brief R grid.
+     */
+    using GridRType = GridR;
+    /**
+     * @brief Theta grid.
+     */
+    using GridThetaType = GridTheta;
+    /**
+     * @brief Batch grid.
+     */
+    using GridBatchType = GridBatch;
+
+    using DimensionRType = typename GridRType::continuous_dimension_type;
+    using DimensionThetaType = typename GridThetaType::continuous_dimension_type;
+
+    using SplineRThetaBuilder = ddc::SplineBuilder2D<
+            ExecutionSpace,
+            typename ExecutionSpace::memory_space,
+            BSplinesR,
+            BSplinesTheta,
+            GridR,
+            GridTheta,
+            ddc::BoundCond::GREVILLE, // boundary at r=0
+            ddc::BoundCond::GREVILLE, // boundary at rmax
+            ddc::BoundCond::PERIODIC,
+            ddc::BoundCond::PERIODIC,
+            ddc::SplineSolver::LAPACK>;
+
+    using SplineRThetaEvaluatorNullBound = ddc::SplineEvaluator2D<
+            ExecutionSpace,
+            typename ExecutionSpace::memory_space,
+            BSplinesR,
+            BSplinesTheta,
+            GridR,
+            GridTheta,
+            ddc::NullExtrapolationRule, // boundary at r=0
+            ddc::NullExtrapolationRule, // boundary at rmax
+            ddc::PeriodicExtrapolationRule<DimensionThetaType>,
+            ddc::PeriodicExtrapolationRule<DimensionThetaType>>;
+
+public:
+    using IdxR = Idx<GridR>;
+    using IdxTheta = Idx<GridTheta>;
+    using IdxBatch = Idx<GridBatch>;
+    using IdxRTheta = Idx<GridR, GridTheta>;
+
+    using IdxRangeR = IdxRange<GridR>;
+    using IdxRangeTheta = IdxRange<GridTheta>;
+    using IdxRangeBatch = IdxRange<GridBatch>;
+    using IdxRangeRTheta = IdxRange<GridR, GridTheta>;
+    using IdxRangeRThetaBatch = IdxRange<GridR, GridTheta, GridBatch>;
+    using IdxRangeBSRTheta = IdxRange<BSplinesR, BSplinesTheta>;
+
+    using DFieldMemRTheta = DFieldMem<IdxRangeRTheta>;
+    using DFieldMemRThetaBatch = DFieldMem<IdxRangeRThetaBatch>;
+    using DFieldMemBSRTheta = DFieldMem<IdxRangeBSRTheta>;
+    using DFieldRTheta = DField<IdxRangeRTheta>;
+    using DFieldRThetaBatch = DField<IdxRangeRThetaBatch>;
+    using DFieldBSRTheta = DField<IdxRangeBSRTheta>;
+    using DConstFieldRTheta = ConstField<double, IdxRangeRTheta>;
+
+    using CoordRTheta = ddc::Coordinate<DimensionRType, DimensionThetaType>;
+
+private:
+    /**
+     * @brief Field of local Larmor radii (rho_L) on the (r, theta) grid.
+     */
+    DFieldRTheta m_rho_L;
+
+public:
+    /**
+     * @brief Constructor.
+     * @param[in] rho_L Field of Larmor radii on the (r, theta) grid.
+     */
+    explicit GyroAverageOperator(DFieldRTheta const& rho_L) : m_rho_L(rho_L) {}
```

Comment:
1. Can we make `rho_L` a DConstField ?
2. Just to make sure, `rho_L` does not evolve in time during a simulation ?

Reply:
Actually, `rho_L` may evolve if background temperature changes? 
I am not quite sure